intellij/src/saros/intellij/editor/EditorManager.java

```diff
@@ -123,109 +121,192 @@ private void execEditorActivity(EditorActivity editorActivity) {
+         * @param path the path to obtain an editor for
+         * @return returns a valid editor for the given resource or <code>null</code> if no such
+         *     editor could be obtained
+         * @see BackgroundEditorPool
+         */
+        private Editor getCalculationEditor(@NotNull SPath path) {
+
+          Editor calculationEditor = editorPool.getEditor(path);
+
+          if (calculationEditor != null) {
+            return calculationEditor;
+          }
+
+          VirtualFile virtualFile = VirtualFileConverter.convertToVirtualFile(path);
+
+          if (virtualFile == null) {
+            log.warn(
+                "Could not create an editor instance for "
+                    + path
+                    + " as no virtual file could be obtained for the resource.");
+
+            return null;
+          }
+
+          Document document = DocumentAPI.getDocument(virtualFile);
+
+          if (document == null) {
+            log.warn(
+                "Could not create an editor instance for "
+                    + virtualFile
+                    + " as no document could be obtained for the file.");
+
+            return null;
+          }
+
+          return backgroundEditorPool.getBackgroundEditor(path, document);
+        }
+
+        /**
+         * Applies the text edit represented by the given values to the given document.
          *
-         * <p>If the given operation is a <code>DeleteOperation</code>, all existing annotations for
-         * the file are adjusted through {@link
-         * AnnotationManager#moveAnnotationsAfterDeletion(IFile, int, int)}.
+         * @param path the resource whose document to modify
+         * @param document the document to modify
+         * @param start the start offset for the modification
+         * @param oldEnd the end offset of the replaced text
+         * @param replacedText the text replaced in the document
+         * @param newText the new text inserted into the document
+         */
+        private void applyTextEdit(
+            @NotNull SPath path,
+            @NotNull Document document,
+            int start,
+            int oldEnd,
+            @NotNull String replacedText,
+            @NotNull String newText) {
+
+          Project project =
+              path.getProject().adaptTo(IntelliJProjectImpl.class).getModule().getProject();
+
+          if (!replacedText.isEmpty()) {
+            String documentReplacedText = document.getText(new TextRange(start, oldEnd));
+
+            if (!replacedText.equals(documentReplacedText)) {
+              log.error(
+                  "Text to be replaced for "
+                      + path
+                      + " from offset "
+                      + start
+                      + " to "
+                      + oldEnd
+                      + " does not match the given replaced text. Should be '"
+                      + StringEscapeUtils.escapeJava(replacedText)
```

Comment:
The main idea behind using `escapeJava` is to escape non-printable characters and avoiding linebreaks in the output? (We cannot assume that the users are writing java, but it should work in most cases)

Reply:
Yes, I mainly used it to escape new lines in the logged content. To be honest, I did not think about that the user might not be using java. I just used the same method already used in other parts of the core for the same purpose (escaping log messages). But, as you already pointed out, it should not matter that much. The method/class is deprecated anyways, so we should replace all of its usages with a unified solution at some point.